test/libsolidity/ASTJSONTest.h

```diff
@@ -41,12 +47,32 @@ class ASTJSONTest: public TestCase
 	void printSource(std::ostream& _stream, std::string const& _linePrefix = "", bool const _formatted = false) const override;
 	void printUpdatedExpectations(std::ostream& _stream, std::string const& _linePrefix) const override;
 private:
+	bool runTest(
+		std::string& _expectation,
+		std::string& _result,
```

Comment:
Any reason why these 2 are not const?

Reply:
Yes, `_result` is filled with the.. result. and on `_expectation` an in-place replacement is run. I suppose we could run the replacement so that it's in a new string.. but I don't see why.